include/os/keyboard.hpp

```diff
@@ -234,12 +234,31 @@ struct combination
     /// Append a combination
     combination & operator+=(const combination &combo)
     {
+        keys.reserve( std::size( keys ) + std::size( combo.keys ) );
```

Comment:
This isn't necessary. `insert()` already does max one allocation here.

Reply:
You are right, I looked insert implementation, so it does as you said. Removed that line and committed